src/android/com/outsystems/plugins/barcodescanner/CustomScannerActivity.java

```diff
@@ -33,17 +29,43 @@
     private boolean flashlightOn=false;
     private boolean isScanning=false;
 
+    // For retrieving R.* resources, from the actual app package
+    // (we can't use actual.application.package.R.* in our code as we
+    // don't know the application package name when writing this plugin).
+    private String package_name;
+    private Resources resources;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_custom_scanner);
 
-        barcodeScannerView = findViewById(R.id.zxing_barcode_scanner);
+        String scanInstructions = getIntent().getStringExtra("SCAN_INSTRUCTIONS");
+        String scanOrientation = getIntent().getStringExtra("SCAN_ORIENTATION");
+        boolean scanLine = getIntent().getBooleanExtra("SCAN_LINE", true);
+        boolean scanButtonVisible = getIntent().getBooleanExtra("SCAN_BUTTON", true);
+        String scanButtonText = getIntent().getStringExtra("SCAN_TEXT");
+
+        //setContentView(R.layout.activity_custom_scanner);
+        //barcodeScannerView = findViewById(R.id.zxing_barcode_scanner);
+        //barcodeScannerView.setTorchListener(this);
+        //switchFlashlightButton = findViewById(R.id.switch_flashlight);
+        //viewfinderView = findViewById(R.id.zxing_viewfinder_view);
```

Comment:
Maybe remove this commented code? Just for maintenance reasons

Reply:
good point, I will remove.